AppServer_Java/src/com/google/appengine/tools/development/ApiServer.java

```diff
@@ -131,6 +133,8 @@ public void close() {
         boolean parsed = response.mergeFrom(post.getResponseBodyAsStream());
         if (!parsed) {
             throw new IOException("Error parsing the response from the HTTP API server.");
+        } else if (response.hasApplicationError()) {
+            throw ApiProxyUtils.getRpcError(packageName, methodName, StatusProto.getDefaultInstance(), response.getApplicationError().getCode(), response.getApplicationError().getDetail());
```

Comment:
I think the following gives equivalent functionality but avoids the additional dependency:

`throw new ApplicationException(response.getApplicationError().getCode(), response.getApplicationError().getDetail());`

Reply:
Wow, that's a huge cleanup. Thanks!